Host Metrics for the Container Agent (#4633)
DataDog/documentation: content/en/agent/faq/host-metrics-with-the-container-agent.md

---
title: Host Metrics with the Container Agent
kind: faq
aliases:
- /agent/faq/getting-further-with-docker/
further_reading:
- link: "/agent/docker/"
  tag: "Documentation"
  text: "Learn more about the Datadog Docker Agent"
---

## Disk check
The [Disk check][1] reports host metrics with the container Agent for any storage mounted on the host and exposed to the container as a volume. No additional setup is required.

### Metrics for partitions
The layered nature of Linux storage (block devices, logical volumes, and partitions) makes it necessary to have a partition mounted to report its free space.

> The container Agent reports disk metrics and rates for every partition that is accessible (even if only partially). This separation is enforced by cgroups and Docker. To allow disk usage reporting on a partition, you need to expose it through a Docker volume with the `-v` argument to `docker run`. The following options are available:

Reviewer: ```suggestion
The container Agent reports disk metrics and rates for every wholly or partially accessible partition. Cgroups and Docker enforce this separation. To allow disk usage reporting on a partition, you need to expose it through a Docker volume with the `-v` argument to `docker run`. The following options are available:
```

* Create a dummy file in the filesystem to watch and expose it through Docker. The Agent can't access any other file on this partition.
    ```
    -v /mnt/loop/dummyfile:/host/loop0:ro
    ```

* Expose the whole mount-point as read-only. The Agent can access the folder hierarchy and world-readable files.
    ```
    -v /mnt/loop:/host/loop0:ro
    ```

* If the mount path is unknown (dynamically mounted volumes), but the parent directory is constant, expose the parent folder. The Agent is able to access all volumes mounted in this folder's children.
    ```
    -v /mnt/:/host/mnt:ro
    ```

### Troubleshooting
#### Missing disk metrics

> If you customized the Docker image or mount a custom directory to the Agent's `conf.d` folder, make sure the Disk check's default `conf.yaml` (or a customized `conf.yaml`) is present, or the check is disabled.

Reviewer: ```suggestion
If you customized the Docker image or mount a custom directory to the Agent's `conf.d` folder, make sure the Disk check's default `conf.yaml` (or a customized `conf.yaml`) is present to enable the check.
```

I am not sure whether this is saying either to verify the conf.yaml or disable the check, or whether it's saying to if there isn't a conf.yaml, it causes the check to be disabled. I assumed the latter with this suggestion.

Author: I will re-word

> If you customized the Docker image or mount a custom directory to the Agent's `conf.d` folder, make sure the Disk check's default `conf.yaml` (or a customized `conf.yaml`) is present, or the check is disabled.

Author: ```suggestion
If you customized the Docker image or mount a custom directory to the Agent's `conf.d` folder, do one of the following for the Disk check:

- Make sure the default `conf.yaml` is present.
- Enable a customized `conf.yaml`.
- Disable the check.
```
